lib/mathutil.js

```diff
@@ -0,0 +1,81 @@
+ * copy of this software and associated documentation files (the "Software"), 
+ * to deal in the Software without restriction, including without limitation 
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, 
+ * and/or sell copies of the Software, and to permit persons to whom the 
+ * Software is furnished to do so, subject to the following conditions:
+ *  
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *  
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING 
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+ * DEALINGS IN THE SOFTWARE.
+ * 
+ */
+
+(function () {
+    "use strict";
+    
+    //source of decimalAdjust from
+    //https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/ceil#Decimal_adjustment
+    /**
+     * Decimal adjustment of a number.
+     *
+     * @param {String}  type  The type of adjustment.
+     * @param {Number}  value The number.
+     * @param {Integer} exp   The exponent (the 10 logarithm of the adjustment base).
+     * @returns {Number} The adjusted value.
+     */
+    function decimalAdjust(type, value, exp) {
+        // If the exp is undefined or zero...
+        if (typeof exp === "undefined" || Number(exp) === 0) {
+            return Math[type](value);
+        }
+        value = Number(value);
+        exp = Number(exp);
+        // If the value is not a number or the exp is not an integer...
+        if (isNaN(value) || !(typeof exp === "number" && exp % 1 === 0)) {
+            return NaN;
+        }
+        // Shift
+        value = value.toString().split("e");
+        value = Math[type](Number(value[0] + "e" + (value[1] ? (Number(value[1]) - exp) : -exp)));
+        // Shift back
+        value = value.toString().split("e");
+        return Number(value[0] + "e" + (value[1] ? (Number(value[1]) + exp) : exp));
+    }
+    
+    function roundLeastMostDecimal(value) {
+        var valueStr = String(value),
+            strLen = valueStr.length,
+            decimalSpot = valueStr.indexOf(".");
+        
+        if (decimalSpot === -1 || decimalSpot === strLen - 2) {
+            return value;
+        }
+        
+        return decimalAdjust("round", value, -(strLen - 2 - decimalSpot));
+    }
+    
+    //does Math.ceil/floor but rounds the least most decimal to try an elimate the most
+    //egregious floating point errors like Math.ciel(100 * .55) === 56 
+    //because 100 * .55 === 55.00000000000001
+    function roundedCeil(value) {
+        return Math.ceil(roundLeastMostDecimal(value));
```

Comment:
Given that we're dealing with calculations on pretty straightforward values, it seems like we could just get away with the round-before-ceil method.  That would avoid having to parse the string value, etc.  example: `Math.ceil((100*.55).toFixed(5))` // 55

Maybe I'm missing some edge cases you've already banged your head against.